qcodes/dataset/sqlite_base.py

```diff
@@ -1332,7 +1334,16 @@ def get_parameter_data(conn: ConnectionPlus,
         param_names = [param.name for param in paramspecs]
         types = [param.type for param in paramspecs]
 
-        res = get_data(conn, table_name, param_names, start=start, end=end)
+        res = get_parameter_tree_values(conn,
+                                        table_name,
+                                        output_param,
+                                        *param_names[1:])
+
+        start = start or 1
+        end = end or len(res)
+
+        res = res[start-1:end]
```

Comment:
I am not too happy with this. It seems pointless to have these start and end arguments if they don't actually limit what you get from the database

Reply:
Perhaps we should just deprecate start/end here

Reply:
I agree.

since it's a relatively new method, we can just remove these arguments :)

alternatively, the start/end part of `get_data` can be factored out and re-used in the query in `get_parameter_tree_values`.

Reply:
The only thing, it is not trivial when to apply start/end - to the whole table or to the de-NULL-ed table?)

Reply:
@astafan8 I think that's why. But as I see it the start and end only have value because they can limit how much data that you load from the db. That does not happen if you load everything and then limit

Reply:
exactly, extend `_build_data_query` and re-use it - what do you think?